CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TreeNode.java

```diff
@@ -0,0 +1,213 @@
+    private static final Logger LOGGER = Logger.getLogger(TreeNode.class.getName());
+
+    private static Map<String, String> cachedHelpMappings = null;
+
+    private final T data;
+
+    private final List<TreeNode<T>> children = new ArrayList<>();
+
+    private TreeNode<T> parent = null;
+
+    public TreeNode(final T data) {
+        this.data = data;
+    }
+
+    public void addChild(final TreeNode<T> child) {
+        child.setParent(this);
+        this.children.add(child);
+    }
+
+    public void addChildren(final List<TreeNode<T>> children) {
+        children.forEach(each -> each.setParent(this));
+        this.children.addAll(children);
+    }
+
+    public List<TreeNode<T>> getChildren() {
+        return children;
+    }
+
+    public T getData() {
+        return data;
+    }
+
+    private void setParent(TreeNode<T> parent) {
+        this.parent = parent;
+    }
+
+    public TreeNode<T> getParent() {
+        return parent;
+    }
+
+    /**
+     * Print the tree to the stringbuilder and return the builder.
+     *
+     * @param <T>
+     * @param node
+     * @param appender
+     * @param builder
+     * @return
+     */
+    public static <T> StringBuilder printTree(final TreeNode<T> node, final String appender, final StringBuilder builder) {
+        builder.append(appender);
+        builder.append(node.getData());
+        node.getChildren().forEach(each -> printTree(each, appender, builder));
+
+        return builder;
+    }
+
+    /**
+     * Write tree to the file writer provided. Caches the helpMappings to
+     * cachedHelpMappings so the recursive calls to write(...) do not call
+     * HelpMapper.getMappings() often
+     *
+     * @param <T> the type of node to store, usually @TOCItem
+     * @param node the tree node which holds the TOCItem information
+     * @param writer the writer to use to write the details of node
+     * @param indent the level of indentation to use when writing
+     */
+    public static <T> void writeTree(final TreeNode<T> node, final FileWriter writer, final int indent) {
+        if (cachedHelpMappings == null) {
+            cachedHelpMappings = HelpMapper.getMappings();
+        }
+
+        write(node, writer, indent);
+    }
+
+    /**
+     * Write tree to the file writer provided. Recurses through children.
+     *
+     * @param <T> the type of node to store, usually @TOCItem
+     * @param node the tree node which holds the TOCItem information
+     * @param writer the writer to use to write the details of node
+     * @param indent the level of indentation to use when writing
+     */
+    private static <T> void write(final TreeNode<T> node, final FileWriter writer, final int indent) {
+        final TOCItem item = (TOCItem) (node.getData());
+        final TreeNode parentNode = node.getParent();
+        final TOCItem parent = parentNode == null ? null : (TOCItem) (node.getParent().getData());
+
+        if ("".equals(item.getTarget()) || item.getTarget() == null) {
```

Comment:
`StringUtils.isBlank(item.getTarget())`

Reply:
Not sure what I was thinking on this one :D